app/helpers/pagelets_helper.rb

```diff
@@ -67,8 +67,70 @@ def th_content(col)
     sort col[:key], as: col[:label], default: col[:default_sort] || 'ASC'
   end
 
+  def mount_column_selector
+    url = "#{api_users_path}/#{User.current[:id]}/table_preferences"
+    has_preference = !@selected_columns.nil?
```

Comment:
I guess it's better to check whether it's `present?`. This will cover `nil` as well as an empty array.
```suggestion
    has_preference = @selected_columns.present?
```